sirfilesystem.c

```diff
@@ -157,6 +157,75 @@ char* _sir_getcwd(void) {
 #endif
 }
 
+#if defined(__OpenBSD__)
+static int
+_sir_openbsdself(char* out, int capacity, int* dirname_length)
+{
+  char   buffer1[4096];
+  char   buffer2[PATH_MAX];
+  char   buffer3[PATH_MAX];
+  char** argv     = (char**)buffer1;
+  char*  resolved = NULL;
+  int    length   = -1;
+
+  for (;;) {
+    int mib[4] = { CTL_KERN, KERN_PROC_ARGS, getpid(), KERN_PROC_ARGV };
+    size_t size;
+
+    if (sysctl(mib, 4, NULL, &size, NULL, 0) != 0) break;
+    if (size > sizeof(buffer1)) {
+      argv = (char**)malloc(size);
+      if (!argv) break;
+    }
+
+    if (sysctl(mib, 4, argv, &size, NULL, 0) != 0) break;
```

Comment:
This is the same call as on line 175... shouldn't you only do it again if something changed? I'm assuming the condition: `size > sizeof(buffer1)`?

Reply:
The first time it doesn't actually get the argument value, but queries and sets the size.  It's a brain dead interface IMO.

Reply:
![IMG_7873](https://github.com/aremmell/libsir/assets/61629094/7b35c182-a2e1-40cc-ba72-b101c8649810)